swarm/network/stream/messages.go

```diff
@@ -152,12 +160,27 @@ func (p *Peer) handleSubscribeMsg(ctx context.Context, req *SubscribeMsg) (err e
 }
 
 type SubscribeErrorMsg struct {
-	Error string
+	Server bool //true indicates message came from upstream, false means downstream
+	Stream Stream
+	Error  string
 }
 
-func (p *Peer) handleSubscribeErrorMsg(req *SubscribeErrorMsg) (err error) {
+func (p *Peer) handleSubscribeErrorMsg(subscribeErr *SubscribeErrorMsg) (err error) {
 	//TODO the error should be channeled to whoever calls the subscribe
-	return fmt.Errorf("subscribe to peer %s: %v", p.ID(), req.Error)
+	if subscribeErr.Server {
```

Comment:
If we rename this to `SourceIsServer`, then this logic is correct, as the source is server thus it is handled on the client, and then the client runs `removeClient`.

We could also rename it to `HandlerIsClient` or something to make it more consistent with `removeClient`....